### PR TITLE
CDMS interface

### PR DESCRIPTION
**Pull request by keflavich**
This is an interface to the relatively simple CDMS database, analogous to the JPL database.

Since these are similar, I reused a lot of code, and in the process made some corrections to jplspec.

https://cdms.astro.uni-koeln.de/cgi-bin/cdmssearch

* [x] TODO: docs!

### CODE REVIEW

**Comment by keflavich** [comment]
@bsipocz I think this is ready to merge now.  Because of the symmetry w/JPLspec, it should be relatively easy.

**Comment by keflavich** [comment]
also we can definitely squash before merging

**Review by bsipocz — COMMENTED** [review]
The main point is that this is a good time to start that namespace restructuring we talked about. 
On place need to be more clear about mandatory or optional arguments, otherwise the rest of the comments are minor to reuse already functionality and do more testing.

**Comment by bsipocz** [comment]
Some squashing of formatting and typo iterations would be welcome, but it doesn't have to be an all in 1

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:13]: yes; let's delay that for now

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:26]: ah yes... I had a docstring in an earlier revision but lost it

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:44]: `astropy.units.Quantity`, right?

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:66]: it's mentioned later; I added a reference

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:98]: I believe `None` is a valid option; it's equivalent to infinite, so I'll document that.

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:103]: agreed

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py:129]: I'm still happy to stick w/80 char line limits when it doesn't hurt.

Also note that this code was directly copied from jplspec

File: astroquery/cdms/core.py
Comment (bsipocz) [reply astroquery/cdms/core.py:98]: Would it make sense then to have it as the default?

Comment (bsipocz) [reply astroquery/cdms/core.py:98]: (If for clarity you rather require a user input in all cases rather than falling back on the default, I'm OK with that, too).

File: astroquery/jplspec/core.py
Comment (keflavich) [reply astroquery/jplspec/core.py:107]: done

File: docs/cdms/cdms.rst
Comment (keflavich) [reply docs/cdms/cdms.rst:1]: ok cool didn't know about that

File: docs/cdms/cdms.rst
Comment (keflavich) [reply docs/cdms/cdms.rst:216]: don't know how to do that, actually!  I was just going off the jplspec example

File: CHANGES.rst
Comment (keflavich) [reply CHANGES.rst:7]: what do you mean?  Should I say 'new module linelists/cdms' ?

Comment (bsipocz) [reply CHANGES.rst:7]: yep, `linelist.cdms`

File: astroquery/cdms/__init__.py
Comment (keflavich) [reply astroquery/cdms/__init__.py]: Done!

File: docs/cdms/cdms.rst
Comment (bsipocz) [reply docs/cdms/cdms.rst:214]: there are some good examples to use the `plot` directive, e.g in photutils:
https://raw.githubusercontent.com/astropy/photutils/main/docs/getting_started.rst

or if the code is standalone rather than part  of the flow, then you can include it with `include_source`, too, see e.g. modeling: https://raw.githubusercontent.com/astropy/astropy/main/docs/modeling/index.rst

File: astroquery/cdms/core.py
Comment (keflavich) [reply astroquery/cdms/core.py]: I think we want users to be explicit here, so we'll leave them as required positional arguments.

File: astroquery/linelists/cdms/core.py
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:13]: This module is not yet moved, so keep the original import
```suggestion
from astroquery.jplspec import lookup_table
```

Comment (keflavich) [reply astroquery/linelists/cdms/core.py:13]: I... didn't make that change.  Huh.

**Comment by keflavich** [comment]
@bsipocz does `..plot ::` inherit from the parent namespace, or does it have to be entirely standalone?

**Comment by bsipocz** [comment]
> @bsipocz does ..plot :: inherit from the parent namespace, or does it have to be entirely standalone?

It's standalone. The photutils example has the workaround to inherit from above, e.g it doesn't show the plot's source code, but has all the lines to  produce  it, but with a skip on the plotting line itself.  Then the plot is standalone and  will  downloadable in the rendered docs (https://photutils.readthedocs.io/en/stable/getting_started.html)

**Comment by bsipocz** [comment]
I'm not sure why it doesn't conflict with the jplspec changes, as those are on master already. Could you maybe do one more rebase, or let me know if I should do it.

File: astroquery/linelists/cdms/core.py
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:131]: I suppose this should be now also be an InvalidQueryError, too to mirror jplspec.

File: astroquery/linelists/cdms/core.py
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:206]: and this one, too to be the new error

**Comment by keflavich** [comment]
ignore current version; I rebased against wrong branch (!)

**Comment by keflavich** [comment]
Ah should I untrack the built .pngs?

**Comment by bsipocz** [comment]
Hmm, yes, I think it is preferred if they don't get tracked unnecessarily.

File: astroquery/linelists/cdms/core.py
Comment (bsipocz) [inline astroquery/linelists/cdms/core.py:108]: ValueError here was more appropriate as at this point there is no query. Sorry about that.

Comment (keflavich) [reply astroquery/linelists/cdms/core.py:108]: I think this is still more informative to users

**Review by bsipocz — APPROVED** [review]

**Comment by keflavich** [comment]
Will merge once we verify the figures are generated correctly

**Comment by keflavich** [comment]
=( images aren't showing up:
https://astroquery--2143.org.readthedocs.build/en/2143/cdms/cdms.html

File: .rtd-environment.yml
Comment (bsipocz) [inline .rtd-environment.yml:14]: good point, we need to add it to the `[docs]` section in setup.cfg, too.

Comment (keflavich) [reply .rtd-environment.yml:14]: done

**Comment by keflavich** [comment]
OK, I fixed up the figures, this is now finally ready to merge.

(last change was an indexing change - most modules are listed in two locations)

**Comment by keflavich** [comment]
@bsipocz do you want this rebased again before merge?

**Comment by bsipocz** [comment]
@keflavich - I did one partial rebase to remove the merge commit, and a trivial additional commit to ensure the downloaded plot scripts indeed work as are, so merging this now.
Thanks!

**Pull request merged**
